### DISCUSSION TITLE
这里页面是n+1的关系吗？

### DISCUSSION THREAD

**Discussion by AshenskyABCDE** [thu-db/huadb#11] [Q&A] [answered]
我是校外的同学，我在做lab1的时候，第一个页面的编号是0，第二个是1，第三个是2...这样吗？
遍历页面的时候是否就是页面号的递增？
在这上面实现理解遇到点问题，希望朋友解答

**Comment by hnjylwb** [comment, marked as answer]
在不考虑 Vacuum 垃圾回收（lab1 进阶功能1）的情况下，页面号就是简单的递增方式。考虑垃圾回收时，如果一张表的中间存在大量垃圾页面，可以通过维护 TablePage 的 next_page_id_ 变量，在扫描表时跳过垃圾页面。因此从通用性的角度来说，建议通过 next_page_id_ 以链表的方式来维护页面。
